cmd/agent/app/agent_test.go

```diff
@@ -147,6 +147,47 @@ func withRunningAgent(t *testing.T, testcase func(string, chan error)) {
 	t.Fatal("Expecting server exit log")
 }
 
+func TestStartStopRace(t *testing.T) {
```

Comment:
I couldn't reproduce this on my mac https://github.com/open-telemetry/opentelemetry-service/issues/43

Reply:
I think `wg.Wait()` doesn't block if `count=0`, not sure if i missed something

Reply:
>I couldn't reproduce this on my mac open-telemetry/opentelemetry-service#43

Yes, unfortunately it is a non-deterministic race, no guarantee that it will be reproducible. The test _may_ catch the problem and it does sometimes when running on my machine.

>I think wg.Wait() doesn't block if count=0, not sure if i missed something

The documentation of WaitGroup.Add() says:

>Note that calls with a positive delta that occur when the counter is zero must happen before a Wait. 

This means it is not valid to call Wait with zero counter before Add is called. However this is precisely what happens in our code.

You are right that in production mode Wait() does not block when count=0 however it is considered a mistake and go test race detector catches this mistake, see this comment https://github.com/golang/go/blob/master/src/sync/waitgroup.go#L124-L127